src/shared/XLinkDevice.c

```diff
@@ -77,8 +77,10 @@ XLinkError_t XLinkInitialize(XLinkGlobalHandler_t* globalHandler)
 {
     XLINK_RET_ERR_IF(pthread_mutex_lock(&init_mutex), X_LINK_ERROR);
     if(init_once){
+        // BUGBUG missing pthread_mutex_unlock()?
         return X_LINK_SUCCESS;
     }
+    // BUGBUG only allow init once? Even if there was an error on previous try?
```

Comment:
+1 - lets refactor to allow calling multiple times until a successful init. Afterwards, can just respond success as here.

Reply:
will do